java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-runtime/src/main/java/com/risingwave/mock/flink/runtime/sinkwriter/SinkWriterV2Impl.java

```diff
@@ -0,0 +1,134 @@
+ * Stateful sink is currently not supported.
+ */
+public class SinkWriterV2Impl<CommT> extends SinkWriterBase {
+
+    org.apache.flink.api.connector.sink2.Sink<RowData> sink;
+    SinkWriter<RowData> writer;
+
+    Committer<CommT> committer;
+
+    public SinkWriterV2Impl(
+            TableSchema tableSchema, org.apache.flink.api.connector.sink2.Sink<RowData> sink) {
+        super(tableSchema);
+        try {
+            this.sink = sink;
+            this.writer = sink.createWriter(new SinkWriterContextV2());
+            if (writer instanceof StatefulSink.StatefulSinkWriter) {
+                throw new UnsupportedOperationException("Don't support StatefulSink");
+            }
+            if (this.sink instanceof TwoPhaseCommittingSink) {
+                this.committer = ((TwoPhaseCommittingSink<?, CommT>) this.sink).createCommitter();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void writeRow(RowData rowData) {
+        try {
+            this.writer.write(
+                    rowData,
+                    new SinkWriter.Context() {
+                        @Override
+                        public long currentWatermark() {
+                            return 0;
+                        }
+
+                        @Override
+                        public Long timestamp() {
+                            return null;
+                        }
+                    });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void write(Iterator<SinkRow> rows) {
+        while (rows.hasNext()) {
+            SinkRow row = rows.next();
+            writeRow(new RowDataImpl(row, getTableSchema().getNumColumns()));
+        }
+    }
+
+    @Override
+    public void sync() {
+        try {
+            this.writer.flush(false);
+            if (committer != null) {
+                Collection<CommT> objects =
+                        ((TwoPhaseCommittingSink.PrecommittingSinkWriter<?, CommT>) this.writer)
+                                .prepareCommit();
+                if (!objects.isEmpty()) {
+                    List<Committer.CommitRequest<CommT>> collect =
+                            objects.stream()
+                                    .map(
+                                            (a) -> {
+                                                return (Committer.CommitRequest<CommT>)
+                                                        new CommitRequestImpl<>(a);
+                                            })
+                                    .collect(Collectors.toList());
+                    committer.commit(collect);
+                }
+            }
+            if (this.writer instanceof TwoPhaseCommittingSink.PrecommittingSinkWriter) {
```

Comment:
We can call `prepareCommit` anyway before checking `if (committer != null)`. When committer not null we only do the commit, and otherwise we should make sure that the objects to commit should be empty.